src/queue.ts

```diff
@@ -63,12 +64,16 @@ export class Queue<TItem> extends EventEmitter {
     function processItem (message : any) : PromiseLike<any> {
       let body = <TItem>JSON.parse(message.Body)
 
-      return Bluebird.resolve(body)
-        .then(handler)
+      return Bluebird.resolve()
+        .then(() => handler(body, message))
```

Comment:
That breaks compatibility with the interface of the library.

If we were to actually do this, I would prefer to pass in an object.

```
return Bluebird.resolve({ item: body, message })
...
```